### PR TITLE
Fix 3 testrunner tests for ie and edge

### PR DESCRIPTION
**Pull request by level420**
1. qx.test.Browser.testForIn for edge: This tests the hash map element order where it seems that edge has the same behavior as all other modern browsers have: sort by hash key.
2. qx.test.ui.embed.Iframe.testSyncSourceAfterDOMMove for IE and edge: IE and edge deliver an extra blank at the end of body.innerText which made the test fail. The extra blank is now deleted specifically for edge and ie.
3. qx.test.io.request.XhrWithRemote."test: progress phases when abort after loading": breaks on win 10 for edge, ie, chrome and firefox. Disabled this test by the added detection for not having win 10 ("noWin10") 
4. qx.test.util.DateForma.testInvalidDate for edge: edge parses even invalid dates and calculates  from the "overflowing" days and months the next "logical" date. In the test the invalid date "2011-11-32" is parsed as "2011-12-02". Disabled this test for edge.

### CODE REVIEW

**Review by johnspackman — DISMISSED** [review]

**Comment by level420** [comment]
Please hold on. Pushing more fixes.

File: framework/source/class/qx/dev/unit/MRequirements.js [hunk 1]
```diff
@@ -256,6 +256,18 @@ qx.Mixin.define("qx.dev.unit.MRequirements", {
...
 
+    /**
+     * Checks if the application is running on Windows 10
+     *
+     * @return {Boolean} <code>false</code> if operating system is Windows 10
+     */
+    hasNoWin10 : function()
+    {
+      var isWin10 = (qx.core.Environment.get("os.name") === "win" && qx.core.Environment.get("os.version") === "10");
+      return (isWin10 ? false : true);
```
Comment (derrell) [inline framework/source/class/qx/dev/unit/MRequirements.js]: `isWin10` is already a boolean. It doesn't need the trinary expression. Could just `return ! isWin10;`

File: framework/source/class/qx/test/util/DateFormat.js [hunk 2]
```diff
@@ -849,6 +858,10 @@ qx.Class.define("qx.test.util.DateFormat",
     dfDE.dispose();
     dfUS.dispose();
 
+  },
+  
+  hasNoEdge: function() {
+    return !(qx.core.Environment.get("browser.name") == "edge");
```
Comment (derrell) [inline framework/source/class/qx/test/util/DateFormat.js]: Any reason not to do `return (qx.core.Environment.get("browser.name") != "edge");` instead? The `!=` operator is easier to read than a negated expression.

**Review by derrell — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/dev/unit/MRequirements.js (hunk 1 above)
Comment (level420) [reply framework/source/class/qx/dev/unit/MRequirements.js]: @derrell please view the other methods in the file. they are all written similar e.g the method `hasNowWin7`.

File: framework/source/class/qx/test/util/DateFormat.js (hunk 2 above)
Comment (level420) [reply framework/source/class/qx/test/util/DateFormat.js]: same here. adopted from https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/test/io/request/XhrWithRemote.js#L227-L229

**Comment by derrell** [comment]
Sigh. Yeah, consistency is important. They should all be fixed.

**Review by derrell — APPROVED** [review]

**Comment by level420** [comment]
Yep, they should maybe be fixed at some point. But for the tests it is not so important I think. 
This PR is for bringing the failing testrunner tests for edge down to 0 to be able to add it as a saucelabs test platform.

**Review by johnspackman — APPROVED** [review]

**Comment by derrell** [comment]
Indeed. That's why I approved it as is. Fixing it globally is a different, probably large, PR.

**Review by cajus — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Comment by derrell** [comment]
Regarding `testForin`, why is this even being tested. Maps are by definition unordered, and you can never, ever, ever count on the order of retrieval. We have that as a known issue in @cboulanger 's Dialog widget (at least in the old version that I'm using), which assumes that retrieving the map keys will occur in the order in which they were generated in a map literal or map element assignment. I really, really hope we don't have any such assumptions within the qooxdoo core, and we should probably remove this test so no one thinks that it's reasonable to assume an order.

**Comment by level420** [comment]
I think we should, if appropriate, delete the `testForin` in a separate PR. The intention of all these PRs is to have safari and edge testrunner succeed again.

**Comment by derrell** [comment]
SGTM.

**Review by cboulanger — APPROVED** [review]

**Comment by johnspackman** [comment]
5 approvals, merging. thank you @level420 !

**Pull request merged**